docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+#### Conform to the protocols provided by the native interface code
+
+Update your native module or component to ensure it implements/extends the native interface that has been code-generated from your JavaScript specs.
+
+Following the example set forth in the previous section, your library might import MyAwesomeSpecs.h, extend the relevant native interface, and implement the necessary methods for this interface:
+
+```objc
+#import <MyAwesomeSpecs/MyAwesomeSpecs.h>
+
+@interface MyAwesomeModule () <StringGetterSpec>
+@end
+
+RCT_EXPORT_METHOD(getString:(NSString *)string
+                   callback:(RCTResponseSenderBlock)callback)
+{
+  // ...
+}
+```
+
+For an existing native module, you will likely already have one or more instances of `[RCT_EXPORT_METHOD](https://reactnative.dev/docs/native-modules-ios#export-a-native-method-to-javascript)`. To migrate to the new architecture, you’ll need to make sure the method signature makes use of the structs provided by the codegen output.
+
+# Supporting the New Architecture in Your Application
+
+## Prerequisites for Supporting the New Architecture in your Application
+
+There’s a few prerequisites that should be addressed before the new architecture is enabled in your application.
+
+### Use a React Native nightly release
+
+At this time, you must use a React Native nightly release in order to get access to the most up to date changes. Eventually, we will recommend targeting a minimum stable open source release.
+
+This guide is written with the expectation that you’re using a specific nightly release. As new revisions of the playbook are release, the target nightly release may be updated. The specific nightly version that we will be using throughout the rest of the playbook is version `0.0.0-20211205-2008-583471bc4`.
+
+Before upgrading your app to a specific nightly release, we recommend upgrading your app to the latest open source release. By upgrading to a published open source release first, you will be able to take advantage of tools like the [upgrade helper](https://react-native-community.github.io/upgrade-helper/) to determine what other changes may be required for your project.
+
+As of this writing, the latest stable release is `0.66.3`. Once you have upgraded your project to this version successfully, you may proceed to targeting the `0.0.0-20211102-2008-8fef52035` nightly release. You may target this nightly release the same way you’d target any other version of React Native:
+
+```bash
+yarn add react-native@0.0.0-20211205-2008-583471bc4
```

Comment:
Question: what is a difference between  `0.0.0-20211205-2008-583471bc4` and `0.0.0-20211215-2008-f16cbe59e`. Could you provide a simple changelog ?

Reply:
I noticed this [file](https://github.com/facebook/react-native/blob/main/scripts/generate-artifacts.js) does not exist in `0.0.0-20211205-2008-583471bc4`

Reply:
Would it be simpler to point to `react-native@nightly` tag then?

Reply:
> Would it be simpler to point to `react-native@nightly` tag then?

agree, it would be great if we could point to react-native@nightly -- the only concern I can think of is nightly build may capture the latest changes that the playbook is not aware of.

Reply:
As a rule of thumb, we tend to run the playbook with a regular cadence. Once we do a run we pick a version of RN native that we will use for the playbook. 

That's true that `react-native@nightly` would also work, but it would create confusion as a user might use a newer version of RN where maybe a script changed and the changes were not reflected (yet) on the playbook.

Till we pin a specific RN stable version (say 0.67.0 or 0.68.0), we prefer to pin to a specific nightly to reduce the variability when running the playbook.

As a side note: if you're willing to do a run on the latest nightly and point out any new problem, that would be very much appreciated 🙏 

> Question: what is a difference between 0.0.0-20211205-2008-583471bc4 and 0.0.0-20211215-2008-f16cbe59e. Could you provide a simple changelog ?

The easiest way would be to run a git log. As you also pointed out, the `generate-artifacts.js` file was missing.